tao_compiler/mlir/disc/utils/source_emitter.cc

```diff
@@ -0,0 +1,601 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Nit: How about naming to 'cuda_source_emitter.h/cpp'

Reply:
One day we may want to add some more types of source emitters and wrapper them in a uniformed data structure.